### PR TITLE
BigUint/BigInt serialization fix

### PR DESCRIPTION
**Pull request by andrei-marinica**
BigUint/BigInt was not using the specialized hooks in some instances.

### CODE REVIEW

File: elrond-wasm-node/src/big_int.rs [hunk 1]
```diff
@@ -246,12 +246,28 @@ impl NestedDecode for ArwenBigInt {
 }
 
 impl TopDecode for ArwenBigInt {
+    const TYPE_INFO: TypeInfo = TypeInfo::BigInt;
+
 	fn top_decode<I: TopDecodeInput>(input: I) -> Result<Self, DecodeError> {
-        Ok(ArwenBigInt::from_signed_bytes_be(&*input.into_boxed_slice_u8()))
+        // since can_use_handle is provided constantly,
+        // the compiler is smart enough to only ever expand one of the if branches
+        let (can_use_handle, handle) = input.try_get_big_int_handle();
```
Comment (camilbancioiu) [inline elrond-wasm-node/src/big_int.rs:254]: Is this the same `try_get_big_int_handle()` you mentioned earlier, in the comment "Unless you're developing elrond-wasm, please ignore."?

File: elrond-wasm-node/src/ext.rs [hunk 2]
```diff
@@ -172,12 +172,28 @@ impl elrond_wasm::ContractHookApi<ArwenBigInt, ArwenBigUint> for ArwenApiImpl {
...
+            bigIntStorageLoadUnsigned(key.as_ref().as_ptr(), key.len() as i32, handle);
+            ArwenBigUint {handle}
         }
     }
 
+    #[inline]
+    fn storage_store_big_uint_raw(&self, key: &[u8], handle: i32) {
+		unsafe {
+            bigIntStorageStoreUnsigned(key.as_ref().as_ptr(), key.len() as i32, handle);
+        }
```
Comment (camilbancioiu) [inline elrond-wasm-node/src/ext.rs:185]: Uneven indentation.

File: elrond-wasm-node/src/ext.rs [hunk 3]
```diff
@@ -172,12 +172,28 @@ impl elrond_wasm::ContractHookApi<ArwenBigInt, ArwenBigUint> for ArwenApiImpl {
...
+	}
+
+	#[inline]
+    fn storage_load_big_uint_raw(&self, key: &[u8]) -> i32 {
+		unsafe {
+            let handle = bigIntNew(0);
+            bigIntStorageLoadUnsigned(key.as_ref().as_ptr(), key.len() as i32, handle);
+            handle
+        }
```
Comment (camilbancioiu) [inline elrond-wasm-node/src/ext.rs:194]: Uneven indentation.

File: elrond-wasm-node/src/ext.rs [hunk 4]
```diff
@@ -456,21 +472,31 @@ impl elrond_wasm::ContractIOApi<ArwenBigInt, ArwenBigUint> for ArwenApiImpl {
...
+	}
+
+    fn get_argument_big_int_raw(&self, arg_id: i32) -> i32 {
+		unsafe {
+            let handle = bigIntNew(0);
+            bigIntGetSignedArgument(arg_id, handle);
+            handle
+        }
```
Comment (camilbancioiu) [inline elrond-wasm-node/src/ext.rs:498]: Indentation.

**Review by sasurobert — DISMISSED** [review]

File: elrond-wasm-node/src/big_int.rs (hunk 1 above)
Comment (andrei-marinica) [reply elrond-wasm-node/src/big_int.rs:254]: Yes

Comment (andrei-marinica) [reply elrond-wasm-node/src/big_int.rs]: It abstracts away the special hook shortcuts, the ones that give the big int handle directly.
We used to do it with TYPE_INFO and if-s everywhere, but it's more elegant this way.

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**